modules/ai/assets/js/editor/ai-layout-behavior.js

```diff
@@ -25,93 +23,24 @@ export default class AiLayoutBehavior extends Marionette.Behavior {
 	onAiButtonClick( e ) {
 		e.stopPropagation();
 
-		this.closePanel();
-
-		this.previewContainer = createPreviewContainer( {
-			// Create the container at the "drag widget here" area position.
+		renderLayoutApp( {
 			at: this.view.getOption( 'at' ),
+			onInsert: this.onInsert.bind( this ),
 		} );
-
-		this.previewContainer.init();
-
-		const rootElement = document.createElement( 'div' );
-		const colorScheme = elementor?.getPreferences?.( 'ui_theme' ) || 'auto';
-		const isRTL = elementorCommon.config.isRTL;
-
-		document.body.append( rootElement );
-
-		ReactDOM.render(
-			<LayoutApp
-				isRTL={ isRTL }
-				colorScheme={ colorScheme }
-				onClose={ () => {
-					this.previewContainer.destroy();
-					this.previewContainer = null;
-
-					ReactDOM.unmountComponentAtNode( rootElement );
-					rootElement.remove();
-
-					this.openPanel();
-				} }
-				onConnect={ onConnect }
-				onGenerate={ this.onGenerate.bind( this ) }
-				onData={ this.onData }
-				onSelect={ this.onSelect.bind( this ) }
-				onInsert={ this.onInsert.bind( this ) }
-			/>,
-			rootElement,
-		);
-	}
-
-	closePanel() {
-		$e.run( 'panel/close' );
-		$e.components.get( 'panel' ).blockUserInteractions();
-	}
-
-	openPanel() {
-		$e.run( 'panel/open' );
-		$e.components.get( 'panel' ).unblockUserInteractions();
 	}
 
 	hideDropArea() {
 		this.view.onCloseButtonClick();
 	}
 
-	onGenerate() {
-		this.previewContainer?.reset();
-	}
-
-	async onData( template ) {
-		const screenshot = await takeScreenshot( template );
-
-		return {
-			screenshot,
-			template,
-		};
-	}
-
-	onSelect( template ) {
-		this.previewContainer.setContent( template );
-	}
-
 	onInsert( template ) {
 		this.hideDropArea();
```

Comment:
isn't hiding the drop area also an editor integration?

Reply:
There is no "Drop area" from a context menu on an existing container